ush/python/pygfs/task/atm_analysis.py

```diff
@@ -0,0 +1,337 @@
+                bkglist.append([os.path.join(rst_dir, basename), os.path.join(run_dir, basename)])
+        for ftype in ['phy_data', 'sfc_data']:
+            template = f'{to_fv3time(self.task_config.current_cycle)}.{ftype}.tile{{tilenum}}.nc'
+            for itile in range(1, task_config.ntiles + 1):
+                basename = template.format(tilenum=itile)
+                bkglist.append([os.path.join(rst_dir, basename), os.path.join(run_dir, basename)])
+
+        bkg_dict = {
+            'mkdir': [run_dir],
+            'copy': bkglist,
+        }
+        return bkg_dict
+
+    @logit(logger)
+    def get_berror_dict(self, config: Dict[str, Any]) -> Dict[str, List[str]]:
+        """Compile a dictionary of background error files to copy
+
+        This method will construct a dictionary of background error files
+        for global atm DA and return said dictionary for use by the FileHandler class.
+
+        Parameters
+        ----------
+        config: Dict
+            a dictionary containing all of the configuration needed
+
+        Returns
+        ----------
+        berror_dict: Dict
+            a dictionary containing the list of atm background error files to copy for FileHandler
+        """
+        if config.STATICB_TYPE in ['bump']:
+            # BUMP atm static-B needs nicas, cor_rh, cor_rv and stddev files.
+            b_dir = config.BERROR_DATA_DIR
+            b_datestr = to_fv3time(config.BERROR_DATE)
+            berror_list = []
+
+            for ftype in ['cor_rh', 'cor_rv', 'stddev']:
+                coupler = f'{b_datestr}.{ftype}.coupler.res'
+                berror_list.append([
+                    os.path.join(b_dir, coupler), os.path.join(config.DATA, 'berror', coupler)
+                ])
+                template = '{b_datestr}.{ftype}.fv_tracer.res.tile{{tilenum}}.nc'
+                for itile in range(1, config.ntiles + 1):
+                    tracer = template.format(tilenum=itile)
+                    berror_list.append([
+                        os.path.join(b_dir, tracer), os.path.join(config.DATA, 'berror', tracer)
+                    ])
+
+            nproc = config.ntiles * config.layout_x * config.layout_y
+            for nn in range(1, nproc + 1):
+                berror_list.append([
+                    os.path.join(b_dir, f'nicas_aero_nicas_local_{nproc:06}-{nn:06}.nc'),
+                    os.path.join(config.DATA, 'berror', f'nicas_aero_nicas_local_{nproc:06}-{nn:06}.nc')
+                ])
+        else:
+            # GSI atm static-B needs namelist and coefficient files.
+            b_dir = os.path.join(config.FV3JEDI_FIX, 'gsibec', config.CASE_ANL)
+            berror_list = []
+            for ftype in ['gfs_gsi_global.nml', 'gsi-coeffs-gfs-global.nc4']:
+                berror_list.append([
+                    os.path.join(b_dir, ftype),
+                    os.path.join(config.DATA, 'berror', ftype)
+                ])
```

Comment:
These should be two separate methods.  The `else` is `gsibec` and not everything.

Reply:
option 1. 
`_get_berror_bump_dict` and `_get_berror_gsibec_dict` would be 2 private methods that will return the be dict.

Then here we can do something like:
```python
berror_dict = SUPPORTED_BERROR_STATIC_MAP['config.STATIC_TYPE']()
```
where `SUPPORTED_BERROR_STATIC_MAP = {'bump': self._get_berror_bump_dict, 'gsibec': self._get_berror_gsibec_dict}`
```

Reply:
@aerorahul , I changed my list methods at [86e63f0](https://github.com/RussTreadon-NOAA/global-workflow/commit/86e63f0900b4645ea8c70664b4007b56cb4498cb) to private dictionary methods.   However, I was unable to get
```
SUPPORTED_BERROR_STATIC_MAP = {'bump': self.__get_berror_bump_dict(self.config), 'gsibec': self.__get_berror_gsibec_dict(self.config)}
berror_dict = SUPPORTED_BERROR_STATIC_MAP['config.STATICB_TYPE']()
```
to work in `get_berror_dict`.   I am not familiar with this syntax so my implementation is wrong.

I did get the following to work
```
        if config.STATICB_TYPE in ['bump']:
            berror_dict = self.__get_berror_bump_dict(config)
        elif config.STATICB_TYPE in ['gsibec']:
            berror_dict = self.__get_berror_gsibec_dict(config)
```
in `get_berror_dict`

This change was committed at [4b4ca4d](https://github.com/RussTreadon-NOAA/global-workflow/commit/4b4ca4d719be1d8bf0a8894ff4f51f7051b997ed).   

I'm willing to change this to your suggestion but I need help to fully implement your suggestion.

Reply:
@aerorahul , through trial and error plus a several google searches I have something which runs to completion.  Not sure if it is what you want.   Also my comments in the berror sections of `atm_analysis.py` probably aren't worded very well.

Changes committed at [3501d59](https://github.com/RussTreadon-NOAA/global-workflow/commit/3501d5990888fc3244b4097b0e96a1ea1e2a634f)

Reply:
@aerorahul :  May I resolve this conversation or is additional work needed in this PR?

Reply:
May this conversation be resolved?